src/google/protobuf/stubs/mathlimits.cc

```diff
@@ -40,42 +40,11 @@
 namespace google {
 namespace protobuf {
 
-// MSVC++ 2005 and older compilers think the header declaration was a
-// definition, and erroneously flag these as a duplicate definition.
-#if defined(COMPILER_MSVC) || __cpluscplus < 201103L
```

Comment:
Thanks for catching this. I think we can just delete the `#if` case and go with the `#else` case, though. Starting from version 3.6 we require C++11 support, so I don't think we need to worry about old versions of MSVC anymore.

Reply:
Well, the `#else` side fails to compile on my machine (Linux/GCC 8.3.1) with redefinition errors. Since everything has always been using the `#if` side since 2015, I think that's the side that should be used.

Reply:
Ah OK, got it, somehow I was thinking the condition would always be false but I see now that it's always true.

Reply:
OK, I've pushed up a change which removes the no-op expansions then.